icic_extraction/grid.py

```diff
@@ -1,31 +1,40 @@
-import re 
+import re
+from tkinter import Y
+from xml.dom.minidom import Element 
 import pandas as pd
 import pdfplumber
+import glob 
+from difflib import SequenceMatcher 
+import pandas as pd
+
 def get_text_from_pdf_with_plumber(apdf):
     tot = ''
     with pdfplumber.open(apdf) as pdf:
         for p in pdf.pages:
             tot += "\n"+p.extract_text()
     return tot 
+x =get_text_from_pdf_with_plumber("/Users/mostafazeiada/Desktop/mostafa/gk_intern_0/icic_extraction/icici_cont/12272021.pdf")
 def get_meta_from_contract(contract_cont):
     """
     This function to get meta data for contract
     and it return alist of basic inf for contract
     """
-    contract_note_no = re.findall("Contract Note No.*?:(.*)",contract_cont)[0]
+    contract_note_no = re.findall("Contract Note No.*?([a-zA-Z0-9]*\/\d+\/\d+)",contract_cont)[0].strip()
     trade_date = re.findall ("TRADE DATE\s*?:\s*(\d{2}-\d{2}-\d{4}).*",contract_cont)[0]
-    settlemnt_no = re.findall ("SETTLEMENT NO.*:\s*(\d+).*",contract_cont)[0]
-    settlement_date = re.findall ("SETTLEMENT DATE.*:\s*(\d{2}-\d{2}-\d{4}).*",contract_cont)[0]
+    settlement_date = re.findall ("SETTLEMENT DATE.*?(\d+-\d+-\d+).*",contract_cont)[0]
     mob_num = re.findall ("Mob No:\s*(\d*).*",contract_cont)[0]
     ucc_no = re.findall("UCC of Client\s*:\s*(\d*)",contract_cont)[0]
```

Comment:
it is better if we have the regex like this : 
UCC\s*of\s*Client as there might be some spaces missing in some documents 
also , it would be better to use re.IGNORECASE